dali/kernels/imgproc/color_manipulation/color_space_conversion_impl.h

```diff
@@ -26,14 +26,20 @@ namespace color {
 
 namespace detail {
 
-template <typename From, typename To>
-constexpr DALI_HOST_DEV float scale_factor() {
-  constexpr double to = is_fp_or_half<To>::value
-                      ? 1.0 : static_cast<double>(max_value<To>());
+template<typename T>
+constexpr DALI_HOST_DEV std::enable_if_t<!is_fp_or_half<T>::value, double> get_max_value() {
+  return static_cast<double>(max_value<T>());
+}
+
+template<typename T>
+constexpr DALI_HOST_DEV std::enable_if_t<is_fp_or_half<T>::value, double> get_max_value() {
+  return 1.0;
+}
 
-  constexpr double from = is_fp_or_half<From>::value
-                        ? 1.0 : static_cast<double>(max_value<From>());
 
+template <typename From, typename To>
+constexpr DALI_HOST_DEV float scale_factor() {
+  constexpr double to = get_max_value<To>(), from = get_max_value<From>();
```

Comment:
The problem here was that `max_value` was not defined for `float16`, because most of `float16`-related functions are non-constexpr. As far as I understand there's no easy solution for that, because `float16` is implementation-defined. Anyway, the `max_value` here was *not needed* for floating types here, so I restructured a code a bit so that `max_value` was not called for floating points at all.